### PR TITLE
net: lib: nrf_cloud: add support for mqtt shadow transform requests

### PR DESCRIPTION
**Pull request by jayteemo**
Support shadow transform (JSONata) requests over MQTT and REST.
IRIS-8721 IRIS-9188

### CODE REVIEW

**Comment by shanthanordic** [comment]
No approvals yet and CI RED moving it out of NCS 2.7.0

**Review by plskeggs — APPROVED** [review]

File: include/net/nrf_cloud_codec.h [hunk 1]
```diff
@@ -125,6 +127,40 @@ struct nrf_cloud_obj_shadow_accepted {
...
 };
 
+struct nrf_cloud_obj_shadow_transform_result {
+	/** The object containing the result of the transform request.
+	 *  The result will have a key of @ref NRF_CLOUD_TRANSFORM_RSP_TF_KEY.
+	 *  The result value type/format depends on the transform request and can be a JSON object,
+	 *  string, number, or boolean.
+	 */
+	struct nrf_cloud_obj obj;
+};
```
Comment (glarsennordic) [inline include/net/nrf_cloud_codec.h:137]: I'm not sold on this needing to be its own struct, but I don't feel super strongly

**Review by glarsennordic — APPROVED** [review]

File: include/net/nrf_cloud_codec.h (hunk 1 above)
Comment (glarsennordic) [reply include/net/nrf_cloud_codec.h:137]: otherwise LGTM

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -809,6 +809,8 @@ Libraries for networking
     * A new nRF Cloud event type ``NRF_CLOUD_EVT_RX_DATA_DISCON`` which is generated when a device is deleted from nRF Cloud.
     * The function :c:func:`nrf_cloud_print_details` to log common nRF Cloud connection information in a uniform way.
     * The :kconfig:option:`CONFIG_NRF_CLOUD_VERBOSE_DETAILS` Kconfig option to enable the :c:func:`nrf_cloud_print_details` function to print all details instead of only the device ID.
+    * Experimental support for shadow transform requests over MQTT using the :c:func:`nrf_cloud_shadow_transform_request` function.
+      Functionality is enabled by the Kconfig option :kconfig:option:`CONFIG_NRF_CLOUD_MQTT_SHADOW_TRANSFORMS`.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:813]:
```suggestion
      This functionality is enabled by the :kconfig:option:`CONFIG_NRF_CLOUD_MQTT_SHADOW_TRANSFORMS` Kconfig option.
```

File: include/net/nrf_cloud.h [hunk 3]
```diff
@@ -845,6 +849,26 @@ int nrf_cloud_send(const struct nrf_cloud_tx_data *msg);
  */
 int nrf_cloud_obj_shadow_update(struct nrf_cloud_obj *const shadow_obj);
 
+/**
+ * @brief Request shadow data over MQTT using a JSONata expression.
+ *        For example, to request the reported device info section, the transform would be:
+ *        "state.reported.device.deviceInfo".
+ *        The application will receive response data in a
```
Comment (peknis) [inline include/net/nrf_cloud.h:856]:
```suggestion
 *        The application will receive response data in an
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**